app/controllers/api/v4/runs_controller.rb

```diff
@@ -33,6 +26,13 @@ def show
         disposition: 'inline',
         filename: "#{@run.filename(timer: timer)}"
       )
+    rescue
+      if params[:historic] == '1'
+        @run.parse(fast: false)
+        render json: @run, serializer: Api::V4::RunWithHistorySerializer
+      else
+        render json: @run, serializer: Api::V4::RunSerializer
+      end
     end
```

Comment:
why the begin/rescue?

Reply:
application/json is not a valid program content type, and with my misunderstanding of the raise from the attribute getter this would throw an exception if you selected application/json, which it then catches and renders the serialzier.